src/vs/platform/userData/common/fileUserDataProvider.ts

```diff
@@ -23,44 +21,63 @@ import { IUriIdentityService } from 'vs/platform/uriIdentity/common/uriIdentity'
  */
 export class FileUserDataProvider extends Disposable implements
```

Comment:
The gist of changes in this file is to be a true wrapper around the passed in disk file system provider and really forward all methods verbatim, with the converted `URI`. Please check on that, it should be simpler now.